exasol/nb_connector/itde_manager.py

```diff
@@ -127,47 +146,60 @@ def _get_ipv4_ddresses():
 
 def is_itde_running(conf: Secrets) -> Tuple[bool, bool]:
     """
-    Checks if the ITDE container exists and if it is running. Returns the two boolean
-    flags - (exists, running).
+    Checks if the ITDE container exists and ready to be used. In the Docker Edition that
+    means the ITDE is running and the AI-Lab container is connected to its network. In
+    other editions it will just check that the ITDE is running.
+
+    Returns two boolean flags - (exists, running).
+
     The name of the container is taken from the provided secret store.
     If the name cannot be found in the secret store the function returns False, False.
     """
 
-    # Try to get the name of the container from the secret store.
+    # Try to get the names of the Docker-DB container and its network from the secret store.
     container_name = conf.get(AILabConfig.itde_container)
-    if not container_name:
+    network_name = conf.get(AILabConfig.itde_network)
+    if not container_name or not network_name:
         return False, False
 
     # Check the existence and the status of the container using the Docker API.
     with ContextDockerClient() as docker_client:
         if docker_client.containers.list(all=True, filters={"name": container_name}):
             container = docker_client.containers.get(container_name)
-            return True, container.status == 'running'
+            is_ready = (container.status == 'running' and not
+                        _is_current_container_not_in_db_network(network_name))
+            return True,  is_ready
         return False, False
 
 
 def start_itde(conf: Secrets) -> None:
     """
-    Starts an existing ITDE container. If the container is already running the function
-    takes no effect. For this function to work the container must exist. If it doesn't
+    Starts an existing ITDE container if it's not already running. In the Docker Edition
```

Comment:
Should this function maybe use the upper `is_itde_running()`?

Reply:
Additionally, I could imagine that some developers might have difficulties distinguishing between the different semantics of `start_itde()` vs. `bring_itde_up()`. Could there be a way to name these functions more accurately?
E.g. renaming `start_itde()` to `restart_itde()`?

Reply:
I apologize as some of my findings probably have been introduced outside the scope of the current pull request.

Reply:
Re: using `is_itde_running`. This function should be renamed to `is_itde_ready`. It now indicates that the ITDE is both running and the AI-Lab container is connected to it. I agree it would be even better to return enum flags, as per you previous comment. In that case, we could call it and then perform actions according to the returned flags.

Re: names. Yes, I agree `restart_itde` would be a better name.

I also agree that these changes go beyond the scope of this PR and, more importantly, will require changes in the upstream AI-Lab code.